### PR TITLE
[EA Forum only] digest ad in the post page footer

### PR DESCRIPTION
**Pull request by s-cheng**
Most of this PR is just moving the homepage RHS digest ad to a separate file. This way I could reuse it in the post page footer. Based on [this comment thread](https://docs.google.com/document/d/1BmbGKMlkFAILwoc9298kEAM5DW2AGKb1ygu9z0vGQ6I/edit?disco=AAAA9u9CuSE).

Screen Shot 2024-01-10 at 9 31 26 PM

-----
Maybe it looks better with a max width:

Screen Shot 2024-01-10 at 9 36 38 PM

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206324629482403) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/ea-forum/DigestAd.tsx
Comment (s-cheng) [inline packages/lesswrong/components/ea-forum/DigestAd.tsx:1]: Most of this is the same with updated class names

File: packages/lesswrong/components/ea-forum/DigestAd.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    // user is logged in and clicked the X in this ad, or "Don't ask again" in the ad in "Recent discussion"
+    currentUser?.hideSubscribePoke
+  )
+  const [loading, setLoading] = useState(false)
+  const [subscribeClicked, setSubscribeClicked] = useState(false)
+  const { flash } = useMessages()
+  const { captureEvent } = useTracking()
+  
+  // Only show this on sites that have a digest
+  if (!hasDigests) return null
```
Comment (s-cheng) [inline packages/lesswrong/components/ea-forum/DigestAd.tsx:136]: Added this check

File: packages/lesswrong/components/ea-forum/DigestAd.tsx [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        Thanks for subscribing! You can edit your subscription via
+        your <Link to={'/account?highlightField=subscribedToDigest'} className={classes.successLink}>
+          account settings
+        </Link>.
+      </div>
+    </div>
+  }
+  
+  return <AnalyticsContext pageSubSectionContext="digestAd">
+    <div className={classNames(classes.root, className)}>
```
Comment (s-cheng) [inline packages/lesswrong/components/ea-forum/DigestAd.tsx:237]: Cleaned up the html a bit

File: packages/lesswrong/components/recommendations/PostBottomRecommendations.tsx [hunk 3]
```diff
@@ -13,6 +14,13 @@ const styles = (theme: ThemeType) => ({
     background: theme.palette.grey[55],
     padding: "60px 0 80px 0",
     marginTop: 60,
+    [theme.breakpoints.down('sm')]: {
+      // make the background flush with the sides of the screen on mobile
+      paddingLeft: 8,
+      paddingRight: 8,
+      marginLeft: -8,
+      marginRight: -8,
+    },
```
Comment (s-cheng) [inline packages/lesswrong/components/recommendations/PostBottomRecommendations.tsx:23]: Also fixed this

**Comment by agnesstenlund** [comment]
screenshots look great! I don't see it on localhost though, do I need to do something else?

**Comment by agnesstenlund** [comment]
(Not sure yet which width I prefer, need to see it more in context)

**Comment by s-cheng** [comment]
@agnesstenlund If you dismiss the ad in one place (like the home RHS) then it's dismissed everywhere, so prob that's why it's not appearing? You can try viewing a post page via incognito to see it

**Review by Will-Howard — APPROVED** [review]
The code looks good 👍 , just a couple of UI things:

1. This button goes slightly higher than the input. I'm not sure if this is intentional but the fact that there is only a tiny difference makes it look off to me
Screenshot 2024-01-12 at 10 56 57

2. On mobile the fonts look a bit to small, especially compare to the section below
Screenshot 2024-01-12 at 10 59 44
3. I think the padding above the ad is a bit too much (it's the same as it was before with just the recommendations, but I think because the ad is smaller it seems like an unusually large gap)

**Comment by agnesstenlund** [comment]
agree with Will it'd be nice if the input was also 40px tall (like the arrow button)

I also tried changing the font to match the other fonts on the bottom of the page, but that also changed the RHS styling, so probably not worth creating a separate thing. But I pushed some padding changes.
Screenshot 2024-01-16 at 14 09 53
Screenshot 2024-01-16 at 14 13 38

Another thing I'd still really like is getting rid of the little * after Email address (which I don't think is a standard pattern, and it should be fairly obvious it's a mandatory input?), as well as making the text black when it's pre-filled for users. I still think the grey text looks so much like a placeholder, I think I'd be confused if I saw my email in there. (Ideally the black text should also be editable so you can use another email, but not a big deal if you can't)

**Comment by agnesstenlund** [comment]
actually, I did a hacky thing where the font sizes will be bigger on medium to smaller screens (since those screens don't have a RHS)

**Comment by s-cheng** [comment]
Ok I've replaced the MUI input with a plain one to remove the * and keep the text black even when it's disabled, and updated the fonts on the post page footer version. Also changed the RHS version to just use the arrow button since the "Subscribe" version was taking up a lot of space.

Screen Shot 2024-01-16 at 11 44 59 AM

Screen Shot 2024-01-16 at 11 48 57 AM

**Comment by agnesstenlund** [comment]
Perfect 🥹

**Pull request merged**